lib/train/transports/azure.rb

```diff
@@ -59,15 +61,17 @@ def platform
         force_platform!('azure', @platform_details)
       end
 
-      def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client)
+      def azure_client(klass = ::Azure::Resources::Profiles::Latest::Mgmt::Client, vault_name = nil)
         if cache_enabled?(:api_call)
           return @cache[:api_call][klass.to_s.to_sym] unless @cache[:api_call][klass.to_s.to_sym].nil?
         end
 
         if klass == ::Azure::Resources::Profiles::Latest::Mgmt::Client
           @credentials[:base_url] = MsRestAzure::AzureEnvironments::AzureCloud.resource_manager_endpoint_url
         elsif klass == ::Azure::GraphRbac::Profiles::Latest::Client
-          client =  GraphRbac.client(@credentials)
+          client = GraphRbac.client(@credentials)
+        elsif klass == ::Azure::KeyVault::Profiles::Latest::Mgmt::Client
+          client = Vault.client(vault_name, @credentials)
```

Comment:
Does Azure Vault support MSI connection? If not we may want to raise somewhere saying its not supported.